src/App.tsx

```diff
@@ -15,48 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = Array<string>;
+
+interface FnArg {
+  sortField: string;
+  reversed: boolean;
+}
+
+enum SortType {
+  sortByAlpha = 'alpha',
+  sortByLength = 'length',
+}
+
+function getPrepearedGoods(goods: Goods, { sortField, reversed }: FnArg) {
+  let prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortType.sortByAlpha:
+          return a.localeCompare(b);
+        case SortType.sortByLength:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
```

Comment:
Rename boolean variables using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isReversed, setIsReversed] = useState(false);
```